contracts/Transfer.sol

```diff
@@ -95,26 +95,41 @@ contract Transfer is FraudProofHelpers {
     function processTransferBatch(
         bytes32 stateRoot,
         bytes memory txs,
-        Types.AccountMerkleProof[] memory accountProofs
+        Types.AccountMerkleProof[] memory accountProofs,
+        uint256 tokenType,
+        uint256 feeReceiver
     ) public pure returns (bytes32, bool) {
         uint256 length = txs.transfer_size();
 
         bool isTxValid;
+        uint256 fees;
+        Tx.Transfer memory _tx;
 
         for (uint256 i = 0; i < length; i++) {
             // call process tx update for every transaction to check if any
             // tx evaluates correctly
+            _tx = txs.transfer_decode(i);
+            fees = fees.add(_tx.fee);
             (stateRoot, , , , isTxValid) = processTx(
                 stateRoot,
-                txs.transfer_decode(i),
+                _tx,
+                tokenType,
```

Comment:
We get this tokenType all the way from the commitment to validate each transaction sender's state leaf has the same token type as the fee receiver's. We can also get the tokenType from the fee receiver's state leaf merkle proof. What do you think?

Reply:
I think it's better to validate the fee receiver's state leaf and then cache the token type, post that we can check the token type of senders against this cached fee receiver's token type. I believe that's what is happening now?

Reply:
Not yet. Let me do this

Reply:
After some trying, I realized this would come with the cost of adding 1 witness in the fraudproof. Detail reasoning in #206. Will stick to committing the tokenType now and defer the decision later.